simbad/rotsearch/__init__.py

```diff
@@ -0,0 +1,719 @@
+
+    @property
+    def rotfun_stdin_template(self):
+        return """TITLE: Generate HKLPCK1 from MODEL FRAGMENT 1
+ROTFUN
+GENE 1   RESO 100.0 {shres}  CELL_MODEL 80 75 65
+CLMN CRYSTAL ORTH  1 RESO  20.0  {shres}  SPHERE   {intrad}
+CLMN MODEL 1     RESO  20.0  {shres} SPHERE   {intrad}
+ROTA  CROSS  MODEL 1  PKLIM {pklim}  NPIC {npic} STEP {step}"""
+
+    @staticmethod
+    def _rot_job_succeeded(amore_z_score):
+        """Check values for job success"""
+        return amore_z_score > 10
+
+    @staticmethod
+    def _mr_job_succeeded(r_fact, r_free):
+        """Check values for job success"""
+        return r_fact < 0.45 and r_free < 0.45
+
+    def rot_succeeded_log(self, log):
+        """Check a rotation search job for it's success
+
+        Parameters
+        ----------
+        log : str
+           The path to a log file
+
+        Returns
+        -------
+        bool
+           Success status of the rot run
+
+        """
+        rot_prog, pdb = os.path.basename(log).replace('.log', '').split('_', 1)
+        rotsearch_parser = simbad.parsers.rotsearch_parser.AmoreRotsearchParser(log)
+        dat_model = [s for s in self.simbad_dat_files if pdb in s][0]
+        score = simbad.rotsearch.amore_score.AmoreRotationScore(
+            pdb, dat_model, rotsearch_parser.alpha, rotsearch_parser.beta, rotsearch_parser.gamma,
+            rotsearch_parser.cc_f, rotsearch_parser.rf_f, rotsearch_parser.cc_i, rotsearch_parser.cc_p,
+            rotsearch_parser.icp, rotsearch_parser.cc_f_z_score, rotsearch_parser.cc_p_z_score,
+            rotsearch_parser.num_of_rot
+        )
+        results = [score]
+        if self._rot_job_succeeded(rotsearch_parser.cc_f_z_score) and pdb not in self.tested:
+            self.tested.append(pdb)
+            output_dir = os.path.join(self.work_dir, "mr_search")
+            mr = simbad.mr.MrSubmit(mtz=self.mtz,
+                                    mr_program=self.mr_program,
+                                    refine_program='refmac5',
+                                    refine_type=None,
+                                    refine_cycles=0,
+                                    output_dir=output_dir,
+                                    tmp_dir=self.tmp_dir,
+                                    timeout=30)
+            mr.mute = True
+            mr.submit_jobs(results,
+                           nproc=1,
+                           process_all=True,
+                           submit_qtype=self.submit_qtype,
+                           submit_queue=self.submit_queue)
+            refmac_log = os.path.join(output_dir, pdb, "mr", self.mr_program, "refine", pdb + "_ref.log")
+            if os.path.isfile(refmac_log):
+                refmac_parser = simbad.parsers.refmac_parser.RefmacParser(refmac_log)
+                return self._mr_job_succeeded(refmac_parser.final_r_fact, refmac_parser.final_r_free)
+        return False
+
+
+class PhaserRotationSearch(object):
```

Comment:
I think both of these classes should be in their own modules ... it makes the code a lot cleaner. If you want to have easier access, add to this file something like:

```python

def AmoreRotationSearch(*args, **kwargs):
    from simbad.rotsearch.amore import AmoreRotationSearch
    return AmoreRotationSearch(*args, **kwargs)


def PhaserRotationSearch(*args, **kwargs):
    from simbad.rotsearch.phaser import PhaserRotationSearch
    return PhaserRotationSearch(*args, **kwargs)
```

Reply:
**Additionally, create an abstract class and use inheritance for the rotation search ... i.e. strip out everything that is duplicated and put it in a higher level object from which you inherit.**